### PR TITLE
Update contributing.md with change log writing guide.

### PR DESCRIPTION
**Pull request by Artemisia-Absinthium**
Updating our contributing documentation after https://github.com/primer/view_components/pull/652#discussion_r657481884

### CODE REVIEW

**Comment by Artemisia-Absinthium** [comment]
Hmmm... I have a failure on the CI checks, but I don't know why it was triggered only now.

https://github.com/primer/view_components/runs/2904761388?check_suite_focus=true#step:4:7

Does any of you know what is happening?

File: docs/content/contributing.md [hunk 1]
```diff
@@ -125,3 +125,23 @@ to the project. Please verify that you have 2FA enabled on both accounts.
 1. Run `script/release` and follow the instructions.
 1. Once your release PR has been approved and merged, run `script/publish`. You may be prompted to log into your rubygem and npm account.
 1. Lastly, draft a new release from the [releases page](https://github.com/primer/view_components/releases). The tag version should be updated to the newest version. The description should be updated to the relevant CHANGELOG descriptions. Press the `Publish release` button and you're good to go!
+
+## Writing changes in CHANGELOG.MD
+This file holds all the changes made in previous versions of Primer View Components and the ones coming to the next version.
+
+To add yours, you need to find in which category to write it under the `main` section. `Main` is the first section on top of the document.
+
+There are three categories currently in use, `Updates`, `Breaking changes` and `Misc`.
```
[the remark this answers is not in the document]
Comment (manuelpuyol) [reply docs/content/contributing.md:135]: I'd add `New` and `Deprecations` too

**Comment by khiga8** [comment]
> Hmmm... I have a failure on the CI checks, but I don't know why it was triggered only now.
> 
> https://github.com/primer/view_components/runs/2904761388?check_suite_focus=true#step:4:7
> 
> Does any of you know what is happening?

@Artemisia-Absinthium It looks like a markdown linter error. Looks like you have trailing spaces in your CHANGELOG.
 
> CHANGELOG.md:42:1 MD009/no-trailing-spaces Trailing spaces [Expected: 0 or 2; Actual: 4]

File: CHANGELOG.md [hunk 2]
```diff
@@ -40,6 +40,10 @@
 
     *Manuel Puyol*
 
+* Update contributing guidelines with change log writting guide.
```
Comment (joelhawksley) [inline CHANGELOG.md:43]:
```suggestion
* Update contributing guidelines with changelog authoring guide.
```

File: docs/content/contributing.md [hunk 3]
```diff
@@ -94,7 +94,7 @@ To minimize the number of restarts, we recommend checking the component in Story
 0. Make sure the tests pass on your machine
 0. Create a new branch: `git checkout -b my-branch-name`
 0. Make your change, add tests, and make sure the tests still pass
-0. Add an entry to the top of `CHANGELOG.md` for your changes
+0. Add an entry to the top of `CHANGELOG.md` for your change. Follow our [writing guideline](https://primer.style/view-components/contributing#writing-changes-in-changelog.md)
```
Comment (joelhawksley) [inline docs/content/contributing.md:97]:
```suggestion
0. Add an entry to the top of `CHANGELOG.md` for your change, [under the appropriate heading](https://primer.style/view-components/contributing#writing-changes-in-changelog.md)
```

File: docs/content/contributing.md [hunk 4]
```diff
@@ -125,3 +125,27 @@ to the project. Please verify that you have 2FA enabled on both accounts.
 1. Run `script/release` and follow the instructions.
 1. Once your release PR has been approved and merged, run `script/publish`. You may be prompted to log into your rubygem and npm account.
 1. Lastly, draft a new release from the [releases page](https://github.com/primer/view_components/releases). The tag version should be updated to the newest version. The description should be updated to the relevant CHANGELOG descriptions. Press the `Publish release` button and you're good to go!
+
+## Writing changes in CHANGELOG.MD
```
Comment (joelhawksley) [inline docs/content/contributing.md:129]:
```suggestion
## Authoring changelog entries
```

File: docs/content/contributing.md [hunk 5]
```diff
@@ -125,3 +125,27 @@ to the project. Please verify that you have 2FA enabled on both accounts.
...
+
+This file holds all the changes made in previous versions of Primer View Components and the ones coming to the next version.
```
Comment (joelhawksley) [inline docs/content/contributing.md:131]: What do you think about putting these guidelines in an HTML comment at the top of the changelog?

Comment (Artemisia-Absinthium) [reply docs/content/contributing.md:131]: Better seeing them twice than never! So yeah, good point :)

As an HTML comment on the Markdown file itself right?

File: docs/content/contributing.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (Artemisia-Absinthium) [reply docs/content/contributing.md]: What about bug fixes?

File: docs/content/contributing.md (hunk 5 above)
Comment (joelhawksley) [reply docs/content/contributing.md:131]: Right!

I think we could probably remove these guidelines from here at that point.

Comment (Artemisia-Absinthium) [reply docs/content/contributing.md]: Yeah, I removed it from the contributing doc to not have to maintain it in two different places.

**Review by manuelpuyol — APPROVED** [review]

**Pull request merged**